includes/Modules/AdSense.php

```diff
@@ -731,25 +685,90 @@ protected function parse_data_response( $method, $datapoint, $response ) {
 				case 'urlchannels':
 					// TODO: Parse this response to a regular array.
 					return $response->getItems();
-				case 'earning-today':
-				case 'earning-yesterday':
-				case 'earning-samedaylastweek':
-				case 'earning-7days':
-				case 'earning-prev7days':
-				case 'earning-this-month':
-				case 'earning-this-month-last-year':
-				case 'earning-28days':
-				case 'earning-prev28days':
-				case 'earning-daily-this-month':
-				case 'earnings-this-period':
-					// TODO: Parse this response to a regular array.
+				case 'earnings':
 					return $response;
 			}
 		}
 
 		return $response;
 	}
 
+	/**
+	 * Gets an array of dates for the given named date range.
+	 *
+	 * @param string $date_range Named date range.
+	 *                           E.g. 'last-28-days'.
+	 *
+	 * @return array|WP_Error Array of [startDate, endDate] or WP_Error if invalid named range.
+	 */
+	private function date_range_to_dates( $date_range ) {
+		switch ( $date_range ) {
+			case 'today':
+				return array(
+					date( 'Y-m-d', strtotime( 'today' ) ),
+					date( 'Y-m-d', strtotime( 'today' ) ),
+				);
+			case 'yesterday':
+				return array(
+					date( 'Y-m-d', strtotime( 'yesterday' ) ),
+					date( 'Y-m-d', strtotime( 'yesterday' ) ),
+				);
+			case 'same-day-last-week':
+				return array(
+					date( 'Y-m-d', strtotime( '7 days ago' ) ),
+					date( 'Y-m-d', strtotime( '7 days ago' ) ),
+				);
+			case '7-days':
+				return array(
+					date( 'Y-m-d', strtotime( '7 days ago' ) ),
+					date( 'Y-m-d', strtotime( 'yesterday' ) ),
+				);
+			case 'prev-7-days':
+				return array(
+					date( 'Y-m-d', strtotime( '14 days ago' ) ),
+					date( 'Y-m-d', strtotime( '8 days ago' ) ),
+				);
```

Comment:
This theoretically can also be accomplished with `$this->parse_date_range( $date_range, 1, 8 )`, but let's keep this as is because it's a special case. We need to revisit the whole date range parsing anyway.

Reply:
Agreed. FWIW I think what's here is more readable than `parse_date_range( $date_range, 1, 8 )`. It would be nice to come up with something more concise without sacrificing readability.